src/ota_metadata/file_table/_orm.py

```diff
@@ -26,28 +29,70 @@
 FT_REGULAR_TABLE_NAME = "ft_regular"
 FT_NON_REGULAR_TABLE_NAME = "ft_non_regular"
 FT_DIR_TABLE_NAME = "ft_dir"
+MAX_ENTRIES_PER_DIGEST = 10
+
+EMPTY_FILE_SHA256 = r"e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855"
+EMPTY_FILE_SHA256_BYTE = bytes.fromhex(
+    "e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855"
+)
 
 
 class FileTableRegularORM(ORMBase[FileTableRegularFiles]):
 
-    _orm_table_name = FT_REGULAR_TABLE_NAME
+    orm_bootstrap_table_name = FT_REGULAR_TABLE_NAME
+    orm_bootstrap_indexes_params = [
+        CreateIndexParams(index_name="digest_index", index_cols=("digest",))
+    ]
+
+    def iter_common_by_digest(
+        self, other_db: str, *, max_entries_per_digest: int = MAX_ENTRIES_PER_DIGEST
+    ) -> Generator[FileTableRegularFiles]:
+        """Yield entries from <other_db>.ft_table which digest presented in this ft.
+
+        This is for assisting faster delta_calculation without full disk scan.
+        """
+
+        attached_db_schema = "base"
+        stmt = f"""\
+        WITH ranked AS (
+            SELECT d2.*, ROW_NUMBER() OVER (PARTITION BY d2.digest) AS rown
+            FROM base.{FT_REGULAR_TABLE_NAME} AS d2
+            INNER JOIN (
+                SELECT digest
+                FROM main.{FT_REGULAR_TABLE_NAME}
+                WHERE digest != {wrap_value(EMPTY_FILE_SHA256_BYTE)}
+                GROUP BY digest
+            ) AS d1 USING(digest) ORDER BY digest
+        )
+        SELECT * FROM ranked WHERE rown <= {max_entries_per_digest};
+        """
```

Comment:
The core sqlite3 query statement to get a list of entries from active_slot's file_table that have digests presented in the OTA image, a.k.a., the files we can re-use.

Reply:
@airkei Please take a careful look at this query :pray: